Assets/js/customer.js

```diff
@@ -249,4 +252,73 @@ document.getElementsByClassName("filter--div")[0].addEventListener('click', func
         initialProducts.sort(compare);
         showCarts();
     }
-}); 
\ No newline at end of file
+}); 
+
+
+// Search bar
+// Name Array
+let nameArray = [];
+initialProducts.map(function(cartItem,i){
```

Comment:
`map` is designed to return a new array, so you could use it that way:
```
const nameArray = initialProducts.map(product => product.productName);
```

Reply:
Yes, I know that.
In my case I need to check if the name is already pushed into the array, so that I can prevent duplicate names